internal/aws/cwlogs/pusher_cache.go

```diff
@@ -0,0 +1,91 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cwlogs // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cwlogs"
+
+import (
+	"context"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+type PusherCache interface {
+	// GetPusher will check to see if there already exists a pusher configured to write to
+	// the given stream and group, and that pusher will be returned.
+	// If no pusher exists for a given stream and group, a new one is created, stored and returned
+	GetPusher(logGroup, logStream string, client Client, retries int) (pusher Pusher)
+
+	// Shutdown will force each stored pusher to flush any data currently buffered
+	// and then clear the existing cache.
+	Shutdown(ctx context.Context) error
+
+	// ListPushers will return a list of all pushers in the cache.
+	ListPushers() []Pusher
+
+	// Flush will cause all pushers to flush to the AWS API
+	Flush() error
+}
+
+type DefaultPusherCache struct {
+	Logger                 *zap.Logger
```

Comment:
This field doesn't need to be exported if an initializer is provided.  Having exported types on a struct like this seems like a thing to avoid.